### ISSUE TITLE
Impossible to fetch node under intersection/union

### ISSUE THREAD

**Issue by elouannf** [google/automotive-design-compose#136]
I tried to reach a node which is used under an intersection in figma. When I tried to fetch it through DesignCompose i couldn't access to it. For example i used the element Modifier and I couln't modifiy my element. Here is a example : 
![image](https://github.com/google/automotive-design-compose/assets/131498280/8e4f3cc4-7fa4-4297-9a8e-f57e920a1fb6)

**Comment by iamralpht** [comment]
Yes, confirming that this is an issue -- children of unions/intersections are always treated as vectors, so the fact that there's a text node is lost. Let's keep this issue to fix customization inside of a union/intersection.

@elouannf we're adding support for mask layers in (https://github.com/google/automotive-design-compose/pull/85)[this change], which will be in the next release (0.21). Would masking let you achieve the effect as a work around?

**Comment by elouannf** [comment]
If masks are added I can use them instead of intersection.  If I can modify the mask thanks to a gauge/ modifier / text element  and acess to nodes inside the mask, the Design should work properly. Thank you for the support can wait to try it out !

**Comment by iamralpht** [comment]
@elouannf we're planning on cutting the RC for 0.21 tomorrow, so you should be able to try that out next week. Alternatively since mask layer support merged into main, you can also checkout main and build everything from source to get an earlier look at the functionality.

**Comment by iamralpht** [comment]
@timothyfroehlich please comment once RC is available

**Comment by iamralpht** [comment]
moving to iteration 0.22 so we keep track of the ticket

**Comment by timothyfroehlich** [comment]
Leaving for vacation distracted me from updating this ticket. RC's for 0.21.0 are available now, and we're wrapping up the full release of 0.21.0 now, which will be available via Maven by the end of the day.

**Comment by elouannf** [comment]
Hi we tried the new release and we found that there is still some trouble with masks ( for instance using Vectors as a mask). are there other limitations ?

**Comment by rylin8** [comment]
Vectors as a mask should be supported. Could you provide an example of a mask setup that isn't working for you? Here are a couple screenshots of masks that we use in our test:
![Screenshot 2023-07-18 at 8 31 49 AM](https://github.com/google/automotive-design-compose/assets/130500397/35663a5c-33bc-4068-b3fa-c31089822833)
![Screenshot 2023-07-18 at 8 33 15 AM](https://github.com/google/automotive-design-compose/assets/130500397/44861c7a-9f7a-4080-bd7a-c59a838c0a1a)

**Comment by iamralpht** [comment]
@elouannf friendly ping on this -- do you have a mask example you can share that isn't working?

**Comment by elouannf** [comment]
Hi Ralph, sorry for the delay I just found the problem for our masks. It's not a big deal because I was able to make it work in another way. But here is the hierarchy that is currently working on figma but not on on DesignCompose it's not very important because I was able to make these changes anyway :
![image](https://github.com/google/automotive-design-compose/assets/131498280/10b1f521-78b7-4a25-94b0-2e955104ab54)
also this one is working in both : 
![image](https://github.com/google/automotive-design-compose/assets/131498280/7f0de72e-572a-43d3-b5c2-b4a9fb8e172f)

**Comment by elouannf** [comment]
[Untitled.zip](https://github.com/google/automotive-design-compose/files/12370866/Untitled.zip)
Hi Ralph , Here is an example. Even if there is an easier way to  create this animation at first and it didn't worked.  I tried to animate using your dial and gauge plugin.
